engine/common/base.h

```diff
@@ -46,3 +46,16 @@ class BaseEmbedding {
 
   // The derived class can also override other methods if needed
 };
+
+class BasePythonModel {
+ public:
+  virtual ~BasePythonModel() {}
+
+  // Model management
+  virtual void Inference(
+      const HttpRequestPtr& req,
+      std::function<void(const HttpResponsePtr&)>&& callback) = 0;
```

Comment:
Having Drogon's dependencies here is against Clean architecture. Models/entities should not be depends on framework